### PR TITLE
Switch from C++11 to C++14

### PR DESCRIPTION
**Pull request by uklotzde**
[As requested on Zulip](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Export.20to.20Engine.20Prime/near/168175313) we should upgrade our build environments and requirements from C++11 to C++14. There is no reason to hold this back any longer!

### CODE REVIEW

**Comment by daschuer** [comment]
Great. 
There is however an issue with Xenial's gcc 4.7 and libstdc++ 4.8. It has only partial c++14 support. We need also to use the c++1y flag in this case.  I am curious to see how far we get in this case.

**Comment by uklotzde** [comment]
Do we really want to support Ubuntu 16.04 for 2.3? Who will take care of this? I recommend **not** to promise support for a platform for which we neither have any resources available, nor a CI build or regular testing in place! Such a promise would be worth nothing and is a misdirection of users.

**Comment by uklotzde** [comment]
I think we have agreed on only supporting only the latest LTS release which is 18.04 for 2.3.

**Comment by uklotzde** [comment]
18.04 comes with Qt 5.9 which means we should also get rid of many Qt workarounds that were needed to support older versions back to Qt 5.2!

**Comment by daschuer** [comment]
I am still on Xenial with my development machine and keeping an eye on compatibility.  

I think we have no official agreement when to drop a Ubuntu release within its lifetime. I think the last common sense was to support it as long there are now good reasons to drop it. We have no c++14 requirements for 2.3 yet. So there is no reason to drop it now. If it is required for a future version with Engine Prime feature, we can cut out a 2.3 branch and enable c++14 in Master. 

I am also fine with a sons flag or auto detect gcc 4.7 and disable Engine Prime in the Xenial case.
For my feeling we have more Xenial users than Engine Prime.

**Comment by daschuer** [comment]
Xenial is on qt 5.5 so we can drop every workaround for smaller versions.

**Comment by uklotzde** [comment]
We are scarce on developer resources! Risking to lose new contributors holding back necessary changes and renovations is not worth it. I'm strongly voting against preserving Xenial support if you are the only person that is able to actually do it.

**Comment by daschuer** [comment]
We do not risk developer resources if we allow c++14 features in master and cut out a 2.3 branch now. 
I think is worth to release 2.3 for Xenial as well because up to now we have no c++14 feature.

File: build/depends.py [hunk 1]
```diff
@@ -443,10 +443,13 @@ def configure(self, build, conf):
...
-        # default but Clang/GCC require a flag.
-        if not build.platform_is_windows:
-            build.env.Append(CXXFLAGS='-std=c++11')
+        # Mixxx requires C++14 support
+        if build.platform_is_windows:
+            # MSVC
+            build.env.Append(CXXFLAGS='/std:c++14')
+        else:
+            # GCC/Clang
+            build.env.Append(CXXFLAGS='-std=c++14')
```
Comment (daschuer) [inline build/depends.py]: it would be fine to check the gcc Version and use -std=c++1y in the gcc < 5.0 case.

**Comment by daschuer** [comment]
Xenial has by default libstdc++6 which includes all c++14 features :-) 
So we just need the flag hack and we can watch how it goes.   
https://en.cppreference.com/w/cpp/compiler_support#cpp14

**Comment by daschuer** [comment]
Sorry for the confusion, until now this branch builds on Xenial 
g++ --version reports 5.4.0 
I have tricked myselfe, because there is also a not installed gcc-4.7 package.

**Comment by uklotzde** [comment]
CI builds succeeded, but have now been restarted due to force push after revert.

**Comment by daschuer** [comment]
LGTM. Thank you and sorry again for the pointless discussion.

**Pull request merged**